BeatSinger/Helpers/Extensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BeatSinger.SimpleJSON;
+
+namespace BeatSinger.Helpers
+{
+    public static class Extensions
+    {
+        public static string GetReasonString(this MusixMatchStatus status)
+        {
+            switch (status)
+            {
+                case MusixMatchStatus.None:
+                    return "Unknown";
+                case MusixMatchStatus.Success:
+                    return "The request was successful.";
+                case MusixMatchStatus.BadRequest:
+                    return "The request had bad syntax or was inherently impossible to be satisfied.";
+                case MusixMatchStatus.Unauthorized:
+                    return "Invalid or missing API key.";
+                case MusixMatchStatus.UsageLimitReached:
+                    return "The usage limit has been reached.";
+                case MusixMatchStatus.Forbidden:
+                    return "You are not authorized to perform this operation.";
+                case MusixMatchStatus.NotFound:
+                    return "The song was not found.";
+                case MusixMatchStatus.MethodNotFound:
+                    return "The requested method was not found.";
+                case MusixMatchStatus.GeneralFailure:
+                    return "A server error occurred.";
+                case MusixMatchStatus.ServiceUnavailable:
+                    return "The MusixMatch service is too busy.";
+                default:
+                    return "Unknown";
+            }
+        }
+
+        public static JSONArray ToJsonArray(this IEnumerable<Subtitle> subtitles)
```

Comment:
I don't like this much, since seeing a `ToJsonArray` in the wild make it seem as if `ToJsonArray` is a generic function that works for all `IEnumerable`s (and that looks like it "pollutes" the namespace).

Reply:
If it makes a difference, the extension only works (and shows up in Intellisense) for `IEnumerable<Subtitle>`. My thinking was `Subtitle` doesn't care how it's being serialized, so I made that method an extension instead of including it in the `Subtitle` class.